### ISSUE TITLE
[Inductor][ROCm] Comprehensive padding can be skipped due to consumer FX-node context

### ISSUE THREAD

**Issue by yoshua2077** [pytorch/pytorch#197869]
### Bug

On ROCm/HIP, TorchInductor can incorrectly skip comprehensive padding for an intermediate tensor because layout freezing uses the current consumer FX node rather than the buffer's producer/origin node.

### Minimal repro

```python
import torch
from torch._inductor import config, metrics

def f(x, y):
    x = torch.matmul(x, y)
    x = x + 1
    return torch.matmul(x, y)

x = torch.randn(2, 16, 2047, device="cuda")
y = torch.randn(2047, 2047, device="cuda")

ref = f(x, y)
metrics.reset()

with config.patch({
    "comprehensive_padding": True,
    "padding_alignment_bytes": 128,
    "padding_stride_threshold": 1024,
    "triton.cudagraphs": False,
}):
    out = torch.compile(f, fullgraph=True)(x, y)

torch.testing.assert_close(ref, out, atol=1e-2, rtol=1e-2)
print("num_comprehensive_padding:", metrics.num_comprehensive_padding)
assert metrics.num_comprehensive_padding > 0
```

Actual:

```text
num_comprehensive_padding: 0
AssertionError
```

The intermediate `[2, 16, 2047]` tensor has stride `[32752, 2047, 1]`. With float32 and 128-byte alignment it should be padded to `[32768, 2048, 1]`.

### Trace

```text
buffer: buf1
producer: add
producer target: aten.add.Tensor
producer dislike_padding: False

current node: mm_1
current target: aten.mm.default
current dislike_padding: True

input stride:  [32752, 2047, 1]
output stride: [32752, 2047, 1]
```

So `_pad_strides()` sees the user-visible consumer `mm_1` and its `dislike_padding=True`, even though the buffer being frozen was produced by `add`.

### A/B check

As a local experiment, running the freeze under the buffer's origin FX node:

```python
origin = buf.get_origin_node()
with V.set_current_node(origin):
    ...
```

changes the result to:

```text
num_comprehensive_padding: 1
PASS
```

### Proposed fix location

`torch/_inductor/ir.py::ExternKernel.convert_to_reinterpret_view()`

This function already gets:

```python
x_unwrap_view_fx_node = buf.get_origin_node()
```

but `freeze_layout()` / `freeze_layout_with_same_order()` still run under the existing `V.current_node`. Using the buffer's origin node as the current node while freezing the layout appears to avoid inheriting `dislike_padding` from an unrelated consumer.

Environment: PyTorch 2.14.0, ROCm/HIP, AMD GPU.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang @naromero77amd @pragupta @jerrymannil @xinyazhang @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben
